kotlin/src/test/kotlin/io/smallrye/mutiny/coroutines/MultiAsFlowTest.kt

```diff
@@ -0,0 +1,295 @@
+        }
+    }
+
+    @Test
+    fun `test immediate completion`() {
+        runBlocking {
+            // Given
+            val multi = Multi.createFrom().empty<Any>()
+
+            // When
+            val eventItems = multi.asFlow().toList()
+
+            // Then
+            assertThat(eventItems).isEmpty()
+        }
+    }
+
+    @Test
+    fun `test failure after emitting an item`() {
+        // Given
+        val emitter = AtomicReference<MultiEmitter<in Int>>()
+        fun emit() = emitter.get().emit(5)
+        fun fail() = emitter.get().fail(IllegalStateException("boom"))
+
+        val multi = Multi.createFrom().emitter { em: MultiEmitter<in Int> ->
+            emitter.set(em)
+        }
+
+        runBlocking {
+            val flow = multi.asFlow()
+            val eventItems = mutableListOf<Int>()
+
+            // When
+            val assertJob = async {
+                assertFails {
+                    flow.collect { eventItems.add(it) }
+                }
+            }
+            GlobalScope.launch {
+                // yield thread and wait shortly for ensuring emitter got set
+                delay(100)
+                emit()
+                emit()
+                fail()
+            }
+
+            // Then
+            assertThat(assertJob.await()).hasMessage("boom").isInstanceOf(IllegalStateException::class.java)
+            assertThat(eventItems).containsExactly(5, 5)
+
+        }
+    }
+
+    @Test
+    fun `test coroutine cancellation before first item`() {
+        // Given
+        val multi = Multi.createFrom().ticks()
+            .startingAfter(Duration.ofMillis(150))
+            .every(Duration.ofMillis(50))
+
+        // When
+        var retrieveError: Throwable? = null
+        runBlocking {
+            val job = launch {
+                try {
+                    multi.asFlow().toList()
+                    fail()
+                } catch (th: Throwable) {
+                    retrieveError = th
+                }
+            }
+            delay(50)
+            job.cancel(CancellationException("abort"))
+        }
+        Thread.sleep(350)
+
+        // Then
+        assertThat(retrieveError).isInstanceOf(CancellationException::class.java)
+    }
+
+    @Test
+    fun `test coroutine cancellation during item delivery`() {
+        // Given
+        val multi = Multi.createFrom().ticks().every(Duration.ofMillis(1))
+        val counter = AtomicInteger()
+
+        // When
+        var retrieveError: Throwable? = null
+        runBlocking {
+            val job = launch {
+                try {
+                    multi.asFlow().collect {
+                        counter.incrementAndGet()
+                    }
+                    fail()
+                } catch (th: Throwable) {
+                    retrieveError = th
+                }
+            }
+            delay(100)
+            job.cancel(CancellationException("abort"))
```

Comment:
So we get a dropped exception there because a call to `onItem` throws an exception:

```
[-- Mutiny had to drop the following exception --]
Exception received by: io.smallrye.mutiny.subscription.Subscribers$CallbackBasedSubscriber.onItem(Subscribers.java:70)
java.util.concurrent.CancellationException: abort
	at io.smallrye.mutiny.coroutines.MultiAsFlowTest$test coroutine cancellation during item delivery$1.invokeSuspend(MultiAsFlowTest.kt:206)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:227)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:166)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:362)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:396)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:388)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeUndispatched(CancellableContinuationImpl.kt:484)
	at kotlinx.coroutines.EventLoopImplBase$DelayedResumeTask.run(EventLoop.common.kt:489)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:84)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.smallrye.mutiny.coroutines.MultiAsFlowTest.test coroutine cancellation during item delivery(MultiAsFlowTest.kt:194)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
[------------------------------------------------]
```